### PR TITLE
[15.0][MIG] purchase_order_general_discount: Migration to version 15.0

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
The general discount field is not positioned correctly, possibly due to the styling of the new tax widget in version 15.

@chienandalu please review the purchase_order_view.xml file to make sure that the general discount field is set correctly.

@Tecnativa TT38331

### CODE REVIEW

File: purchase_order_general_discount/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        ),
+    ]
+
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
+        super().onchange_partner_id()
+        self.general_discount = (
+            self.partner_id.commercial_partner_id.purchase_general_discount
+        )
+        return super().onchange_partner_id()
```
Comment (chienandalu) [inline purchase_order_general_discount/models/purchase_order.py]: This is not the way to solve the pre-commit warning. You're running super() twice here.

```suggestion
        res = super().onchange_partner_id()
        self.general_discount = (
            self.partner_id.commercial_partner_id.purchase_general_discount
        )
        return res
```

**Review by chienandalu — APPROVED** [review]
I don't think we can get much better in a pure xml inheritance approach. We'd need to patch the OWL component and its template to make the perfect fit. Indeed, probably a new module would be needed as a commonground for both purchase and sale general discounts.

**Comment by chienandalu** [comment]
Please review @CarlosRoca13

**Review by chienandalu — CHANGES_REQUESTED** [review]
Sorry I was requesting changes before :)

File: purchase_order_general_discount/models/purchase_order.py (hunk 1 above)
Comment (pilarvargas-tecnativa) [reply purchase_order_general_discount/models/purchase_order.py]: You're right, it's already corrected, thanks for your teachings.

**Comment by pilarvargas-tecnativa** [comment]
> Please review @CarlosRoca13

The changes have been made and you can review it @CarlosRoca13  :)

**Review by CarlosRoca13 — APPROVED** [review]
LGTM :+1:

**Review by chienandalu — APPROVED** [review]
:+1:

**Pull request merged**